src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -409,7 +409,7 @@ internal static void EnsureNullableAttributeExists(CSharpCompilation compilation
                     hasErrors = true;
                 }
             }
-            else if (!defaultExpression.HasAnyErrors && !IsValidDefaultValue(defaultExpression))
+            else if (!defaultExpression.HasAnyErrors && !IsValidDefaultValue(defaultExpression.IsTypelessNew() ? convertedExpression : defaultExpression))
```

Comment:
>!IsValidDefaultValue(defaultExpression.IsTypelessNew() ? convertedExpression : defaultExpression) [](start = 56, length = 97)

Would something break if we simply did `!IsValidDefaultValue(convertedExpression)` here?

Reply:
If we just pass the converted node we report `ERR_DefaultValueMustBeConstant` instead of `ERR_NoConversionForDefaultParam` in some cases.

We need to pass converted node for new() because we want to determine if it's a constant